corehq/sql_db/util.py

```diff
@@ -0,0 +1,102 @@
+from corehq.form_processor.backends.sql.dbaccessors import ShardAccessor
+from corehq.sql_db.config import partition_config
+from django.conf import settings
+
+
+def get_object_from_partitioned_database(model_class, partition_value, partitioned_field_name):
+    """
+    Determines from which database to retrieve a paritioned model object and
+    retrieves it.
+
+    :param model_class: A Django model class
+
+    :param parition_value: The value that is used to partition the model; this
+    value will be used to select the database
+
+    :param partitioned_field_name: The model field on which the object is partitioned; the
+    object whose partitioned_field_name attribute equals partition_value is returned
+
+    :return: The model object
+    """
+    if settings.USE_PARTITIONED_DATABASE:
+        db_name = ShardAccessor.get_database_for_doc(partition_value)
+    else:
+        db_name = 'default'
+
+    kwargs = {
+        partitioned_field_name: partition_value,
+    }
+    return model_class.objects.using(db_name).get(**kwargs)
+
+
+def save_object_to_partitioned_database(obj, partition_value):
+    """
+    Determines to which database to save a partitioned model object and
+    saves it there.
+
+    :param obj: A Django model object
+
+    :param parition_value: The value that is used to partition the model; this
+    value will be used to select the database
+    """
+    if settings.USE_PARTITIONED_DATABASE:
+        db_name = ShardAccessor.get_database_for_doc(partition_value)
+    else:
+        db_name = 'default'
+
+    obj.save(using=db_name)
+
+
+def delete_object_from_partitioned_database(obj, partition_value):
+    """
+    Determines from which database to delete a partitioned model object and
+    deletes it there.
+
+    :param obj: A Django model object
+
+    :param parition_value: The value that is used to partition the model; this
+    value will be used to select the database
+    """
+    if settings.USE_PARTITIONED_DATABASE:
+        db_name = ShardAccessor.get_database_for_doc(partition_value)
+    else:
+        db_name = 'default'
+
+    obj.delete(using=db_name)
+
+
+def run_query_across_partitioned_databases(model_class, q_expression, values=None):
```

Comment:
good question, so far none of our uses of this need paging, and i think that might get a bit tricky since to do paging we need to also sort across dbs. i think it's doable but maybe should go in its own dbaccessor function